Adjust to changes made for selenium

LightOfHeaven1994 opened the pull request:
**Closes** https://issues.redhat.com/browse/IQE-2113

This PR is supposed to remove deprecated items from browserOpts (to avoid failures after bumping selenium to 4.10 - see https://github.com/SeleniumHQ/selenium/pull/12030/files ). I also made small fixes/refactoring of existing code to simplify it a bit.

LightOfHeaven1994 committed:
Adjust to changes made for selenium

RonnyPfannschmidt commented on src/webdriver_kaifuku/__init__.py:
```diff
@@ -143,15 +138,17 @@ def _config_options_for_remote_firefox(browser_conf: dict) -> webdriver.FirefoxO
-        webdriver_kwargs = browser_conf.get("webdriver_options", {})
-        browser_name = _get_browser_name(webdriver_kwargs, webdriver_name)
+        webdriver_kwargs = dict()  # type: Dict[str, Any]
 
```
Please use direct type annotations instead of comments

LightOfHeaven1994 replied:
@RonnyPfannschmidt sure, fixed now

RonnyPfannschmidt reviewed (approved):
Nice change, let's make sure modern typing Syntax is used then this is ready for merge release

RonnyPfannschmidt reviewed (approved):
Thanks

RonnyPfannschmidt commented:
Please Ping me if this isn't released Monday, im Off too a cross country travel and it might slip

LightOfHeaven1994 commented:
@RonnyPfannschmidt I missed some checks so tests failed last time. Fixed tests now (also reverted small part where we were searching for browser name). Can you re-run the pipeline pls?

LightOfHeaven1994 commented:
@RonnyPfannschmidt some tests were failing, now they are fixed. I also reverted a small part of my change so we don't break anyone's existing config. Could you start pipelines again for this MR?

RonnyPfannschmidt commented:
Thanks, started the run again

I think we need to auto approved member's of the org

LightOfHeaven1994 commented:
@RonnyPfannschmidt we are green. You can merge and release it when you have time. There is no rush

RonnyPfannschmidt commented:
Let's quickly see to it then

RonnyPfannschmidt commented:
@LightOfHeaven1994 the release was uploaded, everything seems fine, please give note in case something crops up

thanks for preparing this and have a great week

LightOfHeaven1994 commented:
@RonnyPfannschmidt thanks for reviewing it. Have a great day :)
